airflow/api_connexion/endpoints/dag_endpoint.py

```diff
@@ -100,3 +102,23 @@ def patch_dag(session, dag_id, update_mask=None):
     setattr(dag, 'is_paused', patch_body['is_paused'])
     session.commit()
     return dag_schema.dump(dag)
+
+
+@security.requires_access([(permissions.ACTION_CAN_DELETE, permissions.RESOURCE_DAG)])
+@provide_session
+def delete_dag(dag_id: str, session: Session):
+    """Delete the specific DAG."""
+    # TODO: This function is shared with the /delete endpoint used by the web
+    # UI, so we're reusing it to simplify maintenance. Refactor the function to
+    # another place when the experimental/legacy API is removed.
+    from airflow.api.common.experimental import delete_dag
+
+    try:
+        delete_dag.delete_dag(dag_id)
+    except DagNotFound:
+        raise NotFound(f"Dag with id: '{dag_id}' not found")
+    except DagFileExists:
+        detail = f"Dag with id: '{dag_id}' is still in DagBag. Remove the Dag file first."
+        raise BadRequest(detail=detail)
```

Comment:
I'm don't think this is actually raised any longer? Solves the 400/409 problem at least :)

Reply:
You’re right. I’m also going to remove the block from the web UI view (where I copied the implementation from).